easybuild/tools/build_log.py

```diff
@@ -103,25 +112,45 @@ def deprecated(self, msg, max_ver):
 
     def nosupport(self, msg, ver):
         """Print error message for no longer supported behaviour, and raise an EasyBuildError."""
-        self.error("NO LONGER SUPPORTED since v%s: %s; see %s for more information" % (ver, msg, DEPRECATED_DOC_URL))
+        nosupport_msg = "NO LONGER SUPPORTED since v%s: %s; see %s for more information"
+        raise EasyBuildError(nosupport_msg, ver, msg, DEPRECATED_DOC_URL)
 
     def error(self, msg, *args, **kwargs):
         """Print error message and raise an EasyBuildError."""
-        newMsg = "EasyBuild crashed with an error %s: %s" % (self.caller_info(), msg)
-        fancylogger.FancyLogger.error(self, newMsg, *args, **kwargs)
+        ebmsg = "EasyBuild crashed with an error %s: " + msg
+        args = (self.caller_info(),) + args
+
+        fancylogger.FancyLogger.error(self, ebmsg, *args, **kwargs)
+
         if self.raiseError:
-            raise EasyBuildError(newMsg)
+            self.deprecated("Use 'raise EasyBuildError' rather than error() logging method that raises", '3.0')
+            raise EasyBuildError(ebmsg, *args)
+
+    # note: self is deliberatly ignored
+    def _error_no_raise(self, msg):
+        """Utility function to log an error with raising an exception."""
+        # make sure raising of error is disabled
+        orig_raise_error = self.raiseError
+        self.raiseError = False
+
+        self.deprecated("Use of dedicated _error_no_raise log method", '3.0')
```

Comment:
After giving this some more thought, I don't think we should be using `log.deprecated` at all here...

The idea behind the `log.deprecated` is to indicate that deprecated functionality, i.e. stuff that will be removed in the future, is being triggered.

In this case `_error_no_raise` actually implement the behaviour that we want to be using, so it shouldn't trigger `log.deprecated` at all... Leaving it like this, we'll be triggering `log.deprecated` whenever an `EasyBuildError` is created.

I'll see if I can refurbish things to `log.error` being the behaviour we want (no `raise`), and move the bit currently in `error` that does the `raise` in a separate method that does contain `log.deprecated`.

I'm not sure that will fly though, since by default we actually *want* `log.error` to still raise, for now (to retain backward-compatibility).